packages/utils/interactive-hooks/src/events.types.ts

```diff
@@ -0,0 +1,4 @@
+import { AccessibilityActionEvent, GestureResponderEvent } from 'react-native';
+import { KeyPressEvent } from '.';
+
+export type CallbackEvent = GestureResponderEvent | KeyPressEvent | AccessibilityActionEvent;
```

Comment:
Names are hard 😓if people have a better idea feel free to suggest it!

Reply:
One question I have is: do we have a recommended way for our users to narrow the type?

Reply:
Also, perhaps Input/IOEvent? It's slightly less generic than Callback

Reply:
I think ultimately this should be the `PressEvent` type coming from Pressable. So this intermediate name is fine to me.

Reply:
> One question I have is: do we have a recommended way for our users to narrow the type?

@FalseLobster Narrowing the type would follow whatever you would usually do via TypeScript. Something like `if (e instanceOf GestureResponderEvent)`. I can add an example in a test case.

> Also, perhaps Input/IOEvent? It's slightly less generic than Callback

I think "Input" has its own connotations (it makes me think of typing due to the TextInput RN control). Maybe `InteractionEvent`?

Reply:
Well it turns out that since they're interfaces you can't use "instanceof". So I'm making type guard functions.

Reply:
@FalseLobster Okay type guard functions + unit tests have been added